modules/mono/editor/bindings_generator.cpp

```diff
@@ -5039,24 +5039,6 @@ void BindingsGenerator::_populate_global_constants() {
 			_apply_prefix_to_enum_constants(ienum, prefix_length);
 		}
 	}
-
-	// HARDCODED
-	List<StringName> hardcoded_enums;
-	hardcoded_enums.push_back("Vector2.Axis");
-	hardcoded_enums.push_back("Vector2I.Axis");
-	hardcoded_enums.push_back("Vector3.Axis");
-	hardcoded_enums.push_back("Vector3I.Axis");
-	for (const StringName &enum_cname : hardcoded_enums) {
-		// These enums are not generated and must be written manually (e.g.: Vector3.Axis)
-		// Here, we assume core types do not begin with underscore
-		TypeInterface enum_itype;
-		enum_itype.is_enum = true;
-		enum_itype.name = enum_cname.operator String();
-		enum_itype.cname = enum_cname;
-		enum_itype.proxy_name = pascal_to_pascal_case(enum_itype.name);
-		TypeInterface::postsetup_enum_type(enum_itype);
-		enum_types.insert(enum_itype.cname, enum_itype);
-	}
```

Comment:
The Vector types are intentionally declared manually in C#, so these enums should still be hardcoded. Otherwise, generated API will use `int` instead of the enum types which is worse and also breaks compat.

Reply:
Could the binding generator use the following API to generate this without hardcoding?

https://github.com/godotengine/godot/blob/6206fd5764d99bfbec4e73377428560d3caa0d20/core/variant/variant.h#L800-L804

Also note that `Projection.Planes`, `Vector4.Axis`, and `Vector4i.Axis` are missing.

Reply:
It should be possible yeah, we can just iterate all the Variant types here and then get all their enums and add them to `enum_types`. Something like this (I haven't tested it):

```C++
for (int i = 0; i < Variant::VARIANT_MAX; i++) {
	if (i == Variant::OBJECT) {
		continue;
	}

	Variant::Type type = Variant::Type(i);

	List<StringName> enum_names;
	Variant::get_enums_for_type(type, &enum_names);

	for (const StringName &enum_name : enum_names) {
		TypeInterface enum_itype;
		enum_itype.is_enum = true;
		enum_itype.name = Variant::get_type_name(type) + "." + enum_name;
		enum_itype.cname = enum_itype.name;
		enum_itype.proxy_name = pascal_to_pascal_case(enum_itype.name);
		TypeInterface::postsetup_enum_type(enum_itype);
		enum_types.insert(enum_itype.cname, enum_itype);
	}
}
```

> Also note that `Projection.Planes`, `Vector4.Axis`, and `Vector4i.Axis` are missing.

That's true, these types were added later so it looks like we missed them. That said, it looks like only `Vector3.Axis` is used in Godot APIs, so it's the only one that really matters right now. It would be nice to add the others for future-proofing though.